### PR TITLE
Adding JPY package

### PR DESCRIPTION
**Pull request by IslamAlam**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by IslamAlam** [comment]
@conda-forge/staged-recipes this one is ready for review!

**Comment by IslamAlam** [comment]
@ocefpaf would you please review this PR? It is blocking PR https://github.com/conda-forge/staged-recipes/pull/18709 for the earth observation application.

File: recipes/jpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "jpy" %}
+{% set version = "0.9.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - git_url: https://github.com/jpy-consortium/jpy.git
```
Comment (ocefpaf) [inline recipes/jpy/meta.yaml:16]: Can you use a tarball instead?

**Comment by ocefpaf** [comment]
Please don't remove existing recipes. Our infrastructure takes care of that and, if you do, you can prevent them from making into the feedstock.

File: recipes/jpy/meta.yaml (hunk 1 above)
Comment (IslamAlam) [reply recipes/jpy/meta.yaml:9]: Due to a compatibility issue with SNAP, version v0.9.0 is required. The tarball for v0.9.0 isn’t available on the repository. 
After making sure v0.9.0 is available on conda-forge, I will update it to use the tarball for newer versions.

**Comment by IslamAlam** [comment]
> Please don't remove existing recipes. Our infrastructure takes care of that and, if you do, you can prevent them from making into the feedstock.

It has been resolved.

File: recipes/jpy/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/jpy/meta.yaml:9]: This is a tarball for the same code: https://github.com/jpy-consortium/jpy/archive/refs/tags/0.9.0.tar.gz

You can get the checksum for it and add it as `https://github.com/jpy-consortium/jpy/archive/refs/tags/{{ version }}.tar.gz`.

File: recipes/jpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - wheel
+    - setuptools
+    - numpy
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+
+test:
+  imports:
+  commands:
```
Comment (ocefpaf) [inline recipes/jpy/meta.yaml:44]: Please add some test.

File: recipes/jpy/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+# Install wheel manually
+# $PYTHON -m pip install *.whl
+
+mkdir -p $PREFIX/jpy_wheel
+echo $PY_VER > /tmp/p_ver.txt
+
+VER=$( echo $PY_VER | tr -d '.' )
+
+cp -v $( find $SRC_DIR -name "jpy*$VER*$VER*.whl" ) $PREFIX/jpy_wheel
```
Comment (ocefpaf) [inline recipes/jpy/build.sh:19]: This is a very convoluted way to create a conda-package from a wheel. It seems that a direct `pip install`, as recommended in our docs, would work here, no?

**Comment by IslamAlam** [comment]
@ocefpaf Your feedback is very appreciated. I fixed the issues and did learn more about the build workflow.

**Comment by IslamAlam** [comment]
@conda-forge/staged-recipes This is ready for review

**Comment by IslamAlam** [comment]
@ocefpaf would you please check it out and merge it after reviewing?

**Comment by IslamAlam** [comment]
@dopplershift could you please review this package and merge it?

**Comment by IslamAlam** [comment]
> I guess we need an expert on java here to help you out.

What is needed exactly to be checked?

**Comment by BastianZim** [comment]
Ping @conda-forge/help-java

**Pull request closed without merging**